test/trino.test.ts

```diff
@@ -51,6 +51,56 @@ describe('TrinoFormatter', () => {
     `);
   });
 
+  it('formats row PATTERN()s', () => {
+    const result = format(`
+      SELECT * FROM orders MATCH_RECOGNIZE(
+        PARTITION BY custkey
+        ORDER BY orderdate
+        MEASURES
+                  A.totalprice AS starting_price,
+                  LAST(B.totalprice) AS bottom_price,
+                  LAST(U.totalprice) AS top_price
+        ONE ROW PER MATCH
+        AFTER MATCH SKIP PAST LAST ROW
+        PATTERN ((A | B){5} {- C+ D+ -} E+)
+        SUBSET U = (C, D)
+        DEFINE
+                  B AS totalprice < PREV(totalprice),
+                  C AS totalprice > PREV(totalprice) AND totalprice <= A.totalprice,
+                  D AS totalprice > PREV(totalprice)
+        )
+    `);
+    expect(result).toBe(dedent`
+      SELECT
+        *
+      FROM
+        orders
+      MATCH_RECOGNIZE
+        (
+          PARTITION BY
+            custkey
+          ORDER BY
+            orderdate
+          MEASURES
+            A.totalprice AS starting_price,
+            LAST (B.totalprice) AS bottom_price,
+            LAST (U.totalprice) AS top_price
+          ONE ROW PER MATCH
+          AFTER MATCH
+            SKIP PAST LAST ROW
+          PATTERN
+            ((A | B) {5} {- C + D + -} E +)
+          SUBSET
+            U = (C, D)
+          DEFINE
+            B AS totalprice < PREV (totalprice),
+            C AS totalprice > PREV (totalprice)
+            AND totalprice <= A.totalprice,
```

Comment:
should actually be on one line

```
            C AS totalprice > PREV (totalprice) AND totalprice <= A.totalprice,

```

Reply:
this is a config `logicalOperatorNewline`

Reply:
Ideally this would be on the same line because this is an `AS` statement, but in a `WHERE` clause the operator should be on a new line.

Reply:
Yeah, we'll need to revise this `AND` formatting. Currently it gets applied aggressively everywhere, but it really only makes sense inside WHERE and HAVING.